db/de.go

```diff
@@ -105,7 +105,28 @@ func (d *DEDatabase) AddUserDataUsageBatch(context context.Context, usages map[s
 		placeholders = append(placeholders, "(?::text, ?::bigint)")
 		startargs = append(startargs, usr, usg)
 	}
-	startcte := "WITH new_usages (username, usage) AS (VALUES " + strings.Join(placeholders, ",") + ")"
+	nonzero_usages := "new_nonzero_usages (username, usage) AS (VALUES " + strings.Join(placeholders, ",") + ")"
+
+	// Add a 0 for any user whose most recent total is > 0 but who doesn't appear in the batch we got from the ICAT
+	new_usages2, nuargs, err := psql.Select().
+		Column("us.username").
+		Column("?", 0).
+		From(d.Table("user_data_usage", "udu")).
+		Join(fmt.Sprintf("%s ON (us.id = udu.user_id)", d.Table("users", "us"))).
+		LeftJoin("new_nonzero_usages ON users.username = new_nonzero_usages.username").
```

Comment:
```suggestion
		LeftJoin("new_nonzero_usages ON us.username = new_nonzero_usages.username").
```

I think that `users.username` should refer to the table alias, `us`, instead.